### DISCUSSION TITLE
fish suggestions for git and git sub commands

### DISCUSSION THREAD

**Discussion by ipatch** [fish-shell/fish-shell#10621] [Q&A]
using the latest fish installed via homebrew

```
╰─λ fish --version
fish, version 3.7.1
```

i have a git alias setup in my `~/.gitconfig` to count the number of commits for a git repo, ie. `git cc`

however sometimes i want to use the the actual command the alias is using "under the hood" _so to speak_

and i do not always remember what the command for `git cc` is actually doing. i remember it is something like `git rev ...`

so i type out in my fish prompt `git rev` hit the tab key <kbd>tab</kbd>

but all i see are,

```
╰─λ git rev-list -h
revert  (Revert an existing commit)  rev-parse  (Parse revision names or give repo information)
```

why do i not see a suggestion for `git rev-list` ?

forgive me for my ignorance on this one 👴

also i run `fish_update_completions` from time to time thinking it will help with these sorts of questions, but i'm still getting the above mentioned results.

**Comment by faho** [comment]
The secret is simple: Most sensible completions are hand-written.

Man page generation only gets you so much. To be honest, we have so many scripts now that it barely does anything anymore.

And the git completions lacked rev-list, which I've [now added](https://github.com/fish-shell/fish-shell/commit/569d3cdfffce7b7b163375844ec944989a497f1a). Git is absolutely fudging *GINORMOUS*, and writing good git completions could be a full-time job for a good six months or so.
